uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-request-action-dropdown.spec.js

```diff
@@ -0,0 +1,1541 @@
+/* eslint-env amd, jasmine */
+
+define([
+  'common/angular',
+  'common/lodash',
+  'common/moment',
+  'mocks/data/option-group-mock-data',
+  'mocks/data/absence-type-data',
+  'mocks/data/leave-request-data',
+  'leave-absences/shared/config',
+  'leave-absences/manager-leave/app'
+], function (angular, _, moment, optionGroupMock, absenceTypeData, leaveRequestData) {
+  'use strict';
+
+  describe('leaveRequestActionDropdown', function () {
+    var $componentController, $log, $q, $rootScope, vm, LeaveRequestInstance, dialog, sharedSettings, role, request;
+    var absenceTypes = _.indexBy(absenceTypeData.all().values, 'id');
+    var requestStatuses = _.indexBy(optionGroupMock.getCollection('hrleaveandabsences_leave_request_status'), 'value');
+
+    beforeEach(module('manager-leave'));
+
+    beforeEach(inject(['shared-settings', function (_sharedSettings_) {
+      sharedSettings = _sharedSettings_;
+    }]));
+
+    beforeEach(inject(function (_$componentController_, _$log_, _$q_, _$rootScope_, _dialog_, _LeaveRequestInstance_) {
+      $componentController = _$componentController_;
+      $log = _$log_;
+      $q = _$q_;
+      $rootScope = _$rootScope_;
+      dialog = _dialog_;
+      LeaveRequestInstance = _LeaveRequestInstance_;
+    }));
+
+    beforeEach(function () {
+      spyOn($log, 'debug');
+    });
+
+    beforeEach(function () {
+      role = 'staff';
+      request = LeaveRequestInstance.init(leaveRequestData.all().values[0], true);
+
+      compileComponent();
+    });
+
+    it('is initialized', function () {
+      expect($log.debug).toHaveBeenCalled();
+    });
+
+    describe('for admin', function () {
+      beforeEach(function () {
+        role = 'admin';
+        request = getRequestByType('leave');
+
+        compileComponent();
+      });
+
+      it('has correct role', function () {
+        expect(vm.role).toBe(role);
+      });
+
+      describe('for Leave request', function () {
```

Comment:
As discussed there is massive unnecessary repetition in this file, please refactor the tests as something like
```js
describe('when the leave request status is "awaiting approval"', function () {
  describe('when the user is an admin', function () {
    it('shows the actions X,Y,Z', function () {
    });
  });
  describe('when the user is a manager', function () {
    it('shows the actions X,Y', function () {
    });
  });
  describe('when the user is a staff', function () {
    it('shows the actions X', function () {
    });
  });
});
```